### PR TITLE
Feature/track uploads cx 1550

### PR DESCRIPTION
**Pull request by hjheath**
I can see that I am sending the upload method and duration in ms to woopra.

I have created woopra fuunels to aggregate the speeds, but they seem to work on a per user basis so it's hard to test if they are working properly.

### CODE REVIEW

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-279-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-279-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://279-pr-onfido-sdk-ui-onfido.surge.sh

File: src/components/Confirm/index.js [hunk 1]
```diff
@@ -163,6 +163,8 @@ class Confirm extends Component  {
   }
 
   onApiSuccess = (apiResponse) => {
+    const duration = Date.now() - this.startTime
```
Comment (rfreitas) [inline src/components/Confirm/index.js]: so why not track the spinner instead of the api calls?

Comment (hjheath) [reply src/components/Confirm/index.js]: We use the spinner in several places so we would need to pass the action (e.g. document upload) to the spinner to track uploads reliably.

I could pull out the tracking and logic display the spinner so they are explicitly coupled, within the `Confirm` component, but that feels like overkill to me. What do you think?

File: src/components/Confirm/index.js [hunk 2]
```diff
@@ -175,8 +177,10 @@ class Confirm extends Component  {
   }
 
   uploadCaptureToOnfido = () => {
-    this.setState({uploadInProgress: true})
     const {validCaptures, method, side, token} = this.props
+    this.startTime = Date.now()
```
Comment (stefaniacardenas) [inline src/components/Confirm/index.js]: Isn't `performance.now()` better suited for these kind of measurements?

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-279-pr-onfido-sdk-ui-onfido.surge.sh

File: src/components/Confirm/index.js (hunk 1 above)
Comment (rfreitas) [reply src/components/Confirm/index.js]: An event gets its `step` prepended if you call `this.trackScreen`

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-279-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://279-pr-onfido-sdk-ui-onfido.surge.sh

File: src/components/Confirm/index.js [hunk 3]
```diff
@@ -163,6 +163,8 @@ class Confirm extends Component  {
   }
 
   onApiSuccess = (apiResponse) => {
+    const duration = Math.round(performance.now() - this.startTime)
+    sendEvent('Upload complete', {duration, method: this.props.method})
```
Comment (hjheath) [inline src/components/Confirm/index.js]: @rfreitas I tried tracking the Spinner screen using the `trackComponent` method. However this did not give accurate timings on woopra (see screenshot). Therefore I think we do need to explicitly time how long the upload takes and send the duration we measure to woopra. I think it makes sense to keep this logic in the Confirm component so I think we should leave this PR how it is.

screen shot 2017-12-11 at 11 54 44

Comment (hjheath) [reply src/components/Confirm/index.js]: Compare explicitly timing the upload:
screen shot 2017-12-11 at 11 59 50

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-279-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-279-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://279-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by seewah** [comment]
maybe I did not pick this up last time, and apologies if did not, but a minor point
we have
- `starting upload`
- `upload complete`
- `completed flow`

would it not be better to call `upload complete` `completed upload` instead for some sort of consistency?

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-279-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-279-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://279-pr-onfido-sdk-ui-onfido.surge.sh

**Pull request merged**